### PR TITLE
i18n(ko): Translated `astro-components.mdx`

### PR DESCRIPTION
**Pull request by minjongbaek**
#### What kind of changes does this PR include?

- New or updated content
- Translated content

#### Description

- Translated `astro-components.mdx` to korean

### CODE REVIEW

File: src/content/docs/ko/core-concepts/astro-components.mdx [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+i18nReady: true
+---
+
+**Astro 컴포넌트**는 모든 Astro 프로젝트의 기본 구성 요소로서 클라이언트 측 런타임이 없는 HTML 전용 템플릿 컴포넌트입니다. Astro 컴포넌트는 `.astro` 파일 확장자로 식별할 수 있습니다.
+
+Astro 컴포넌트는 매우 유연합니다. 대부분의 경우 Astro 컴포넌트는 헤더나 프로필 카드와 같이 **페이지에서 재사용할 수 있는 가능한 UI**를 포함할 수 있습니다. 또한 SEO 작업을 쉽게 해주는 일반적인 `<meta>` 태그 모음과 같은 작은 HTML 스니펫을 포함할 수도 있습니다. Astro 컴포넌트는 전체 페이지 레이아웃을 포함할 수도 있습니다.
+
+Astro 컴포넌트에 대해 알아야 할 가장 중요한 점은 **클라이언트에서 렌더링 되지 않는다는 것**입니다. 빌드 시점 혹은 [server-side-rendering (SSR)](/ko/guides/server-side-rendering/)을 통해 온디맨드 방식으로 HTML로 렌더링 됩니다. 컴포넌트 프론트매터에 자바스크립트 코드를 포함할 수 있으며, 사용자의 브라우저로 전송되는 최종 페이지에서 모든 코드가 제거됩니다. 그 결과 기본적으로 자바스크립트 풋프린트가 전혀 추가되지 않아 사이트 속도가 빨라집니다.
+
+Astro 컴포넌트에 클라이언트 측 상호 작용이 필요한 경우에는 [standard HTML `<script>` tags](/ko/guides/client-side-scripts/)나 [UI Framework components](/ko/core-concepts/framework-components/#hydrating-interactive-components)를 추가할 수 있습니다. 자세한 내용은 다음 링크에서 확인하세요.
```
Comment (sujang958) [inline src/content/docs/ko/core-concepts/astro-components.mdx:13]: 이게 더 자연스러운 것 같아요
```suggestion
Astro 컴포넌트에 클라이언트 측 상호 작용이 필요한 경우에는 [standard HTML `<script>` tags](/ko/guides/client-side-scripts/)나 [UI Framework components](/ko/core-concepts/framework-components/#hydrating-interactive-components)를 추가할 수 있습니다. 자세한 내용은 링크에서 확인하세요.
```

**Review by sujang958 — APPROVED** [review]
LGTM!

File: src/content/docs/ko/core-concepts/astro-components.mdx [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+// 컴포넌트 스크립트 (JavaScript)
+---
+<!-- 컴포넌트 템플릿 (HTML + JS Expressions) -->
+```
+
+### 컴포넌트 스크립트
+
+Astro는 코드 펜스(`---`)를 사용하여 Astro 컴포넌트에서 컴포넌트 스크립트를 식별합니다. 이전에 마크다운을 작성해 본 적이 있다면 *프론트매터*라는 비슷한 개념에 이미 익숙할 것입니다. Astro의 컴포넌트 스크립트 아이디어는 이 개념에서 직접 영감을 받았습니다.
+
+컴포넌트 스크립트를 사용하여 템플릿을 렌더링하는 데 필요한 모든 JavaScript 코드를 작성할 수 있습니다. 여기에는 다음이 포함될 수 있습니다.
```
Comment (WooDaeHyun) [inline src/content/docs/ko/core-concepts/astro-components.mdx:31]:
```suggestion
컴포넌트 스크립트를 사용하여 템플릿을 렌더링하는 데 필요한 모든 JavaScript 코드를 작성할 수 있습니다. 여기에는 다음과 같은 것들이 포함될 수 있습니다.
```

로 수정해주면 더 의미전달이 잘 될 것 같아요! 👍

**Review by WooDaeHyun — COMMENTED** [review]
hello! I think this page was translated almost perfectly! 

super cool! 👍

**Review by WooDaeHyun — APPROVED** [review]

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Great work peeps 💜🚀

**Pull request merged**